src/elements/content-explorer/ContentExplorer.js

```diff
@@ -383,6 +391,52 @@ class ContentExplorer extends Component<Props, State> {
         }
     };
 
+    /**
+     * Format template, representation, and tokenString into a url usable in the DOM
+     *
+     * @param {string} representation - representation field from representations field of item
+     * @param {string} template - template field from representations field of item
+     * @param {string} tokenString - string of access_token
+     * @return {string} - generated thumbnailUrl
+     */
+    makeThumbnailUrl(representation: string, template: string, tokenString: string): string {
+        const thumbnailBaseUrl = template.replace('{+asset_path}', representation === 'jpg' ? '' : '1.png');
+        return `${thumbnailBaseUrl}?access_token=${tokenString}`;
+    }
+
+    /**
+     * Updates item in currentCollection with the given thumbnailUrl
+     *
+     * @param {BoxItem} item - item that should have thumbnailUrl field set
+     * @param {?string} representation - representation from item's representations field
+     * @param {?string} template -  template from item's representations field
+     * @return {Promise<void>}
+     */
+    assignItemThumbnailUrl = async (item: BoxItem, representation: ?string, template: ?string): Promise<void> => {
+        const { token } = this.props;
+        if (!!template && !!representation) {
+            const tokenResult = await TokenService.getToken(getTypedFileId(item.id), token);
+            if (
```

Comment:
I don't think just checking `!tokenResult` would be sufficient, since `getToken` can return an object without a `.read` field.  (If the return token object only has a `.write` field, I'm not sure that we can use that to get the thumbnails.)  I was able to simplify the logic though.